tests/test_prepare_dutchf3.py

```diff
@@ -0,0 +1,288 @@
+    SLICE_STEPS = 1
+
+    # use a temp dir that will be discarded at the end of the execution
+    with tempfile.TemporaryDirectory() as tmpdirname:
+
+        # saving the file to be used by the script
+        label_file = tmpdirname + '/label_file.npy'
+        np.save(label_file, ALINE)
+
+        # stting the output directory to be used by the script
+        output = tmpdirname + '/split'
+
+        # calling the main function of the script without SLICE_STEPS, to check default value
+        prep_dutchf3.split_section_train_val(data_dir=tmpdirname, output_dir=output, label_file=label_file,slice_steps=SLICE_STEPS, per_val=PER_VAL, log_config=LOG_CONFIG)
+
+        # reading the file and splitting the data
+        section_train = pd.read_csv(output + '/section_train.txt', header=None, names=['row'])
+        section_train = pd.DataFrame(section_train.row.str.split('_').tolist(), columns=['aline', 'section'])
+
+        section_val = pd.read_csv(output + '/section_val.txt', header=None, names=['row'])
+        section_val = pd.DataFrame(section_val.row.str.split('_').tolist(), columns=['aline', 'section'])
+
+        # test
+        assert (float(section_train.section[1]) - float(section_train.section[0])) % float(SLICE_STEPS) == 0.0
+        assert (float(section_val.section[1]) - float(section_val.section[0])) % float(SLICE_STEPS) == 0.0
+
+def test_prepare_dutchf3_section_step_2():
+
+    """check a complete run for the script in case further changes are needed
+    """
+    # setting a value to SLICE_STEPS as needed to test the values
+    SLICE_STEPS = 2
+
+    # use a temp dir that will be discarded at the end of the execution
+    with tempfile.TemporaryDirectory() as tmpdirname:
+
+        # saving the file to be used by the script
+        label_file = tmpdirname + '/label_file.npy'
+        np.save(label_file, ALINE)
+
+        # stting the output directory to be used by the script
+        output = tmpdirname + '/split'
+
+        # calling the main function of the script without SLICE_STEPS, to check default value
+        prep_dutchf3.split_section_train_val(data_dir=tmpdirname, output_dir=output, label_file=label_file,
+                                                slice_steps=SLICE_STEPS, per_val=PER_VAL, log_config=LOG_CONFIG)
+
+        # reading the file and splitting the data
+        section_train = pd.read_csv(output + '/section_train.txt', header=None, names=['row'])
+        section_train = pd.DataFrame(section_train.row.str.split('_').tolist(), columns=['aline', 'section'])
+
+        section_val = pd.read_csv(output + '/section_val.txt', header=None, names=['row'])
+        section_val = pd.DataFrame(section_val.row.str.split('_').tolist(), columns=['aline', 'section'])
+
+        # test
+        assert (float(section_train.section[1]) - float(section_train.section[0])) % float(SLICE_STEPS) == 0.0
+        assert (float(section_val.section[1]) - float(section_val.section[0])) % float(SLICE_STEPS) > 0.0
```

Comment:
can you please explain why the the test requires the result to be 0 in the train case, and > 0 in the val case?

Reply:
I only skip lines to the training set, and do nothing to validation set. That's why the test is like this.

This was as per the original issue opened [here](https://github.com/olgaliak/seismic-deeplearning/issues/9#issuecomment-582650229)